perceval/backends/core/github.py

```diff
@@ -662,6 +683,23 @@ def pull_review_comments(self, pr_number):
         comments_url = urijoin("pulls", str(pr_number), "comments")
         return self.fetch_items(comments_url, payload)
 
+    def pull_commit_comments(self, commit_sha):
+        """Get pull request commit comments"""
+
+        if commit_sha in self._commit_comments:
+            return self._commit_comments[commit_sha]
+
+        payload = {
+            'per_page': PER_PAGE,
+            'direction': 'asc',
+            'sort': 'updated'
+        }
+        path = urijoin("commits", commit_sha, "comments")
+
+        items = [items for items in self.fetch_items(path, payload)]
```

Comment:
I tried it earlier and found that  `self.fetch_items(path, payload)` returns a generator object. Then the code `for items in self.fetch_items(path, payload)` gets a string which is a list of commit comments. So items[0] is just a string which comprises of the list of commit_comments.

Reply:
I also was not able to find much for commit comments, so used this : [https://github.com/octocat/Spoon-Knife/pull/1176](https://github.com/octocat/Spoon-Knife/pull/1176) for the example.
The required pull_request (1176) data is quickly retrieved using the command`perceval github octocat Spoon-Knife --from-date "2019-03-06T23:25:42Z" --category pull_request --sleep-for-rate -t '3f7210425d61134182aa5c606079d1c63c651cee'>> testing.json 
`

Reply:
> I tried it earlier and found that self.fetch_items(path, payload) returns a generator object. Then the code for items in self.fetch_items(path, payload) gets a string which is a list of commit comments. So items[0] is just a string which comprises of the list of commit_comments.

If I'm not wrong the method `fetch_items` returns paginated items so `items[0]` should return just the first page (I'm going to check to confirm it)

> I also was not able to find much for commit comments, so used this : octocat/Spoon-Knife#1176 for the example...

Were you able to create one or more commit comments in a review ?

Reply:
I tried again to create a commit comment within a review but I didnt' find how. If we are not able to create one, it's possible that this feature is not supported anymore by GitHub (I'm really sorry about that @harshalmittal4). In this case, we should probably freeze the PR and contact GitHub, or see how we can reuse this code. What do you think @harshalmittal4 (any idea is welcomed)?

Reply:
hey @valeriocos, some views:
1) I am not able to figure out currently why the commit comments box at the end is showing up in some PR's but not in others, maybe github has discontinued that recently or it might be some other case (Since [this](https://github.com/yashkant/Padam-Tensorflow/pull/9/commits/095d5dcbccb4e1f28cc9b788c470bbc308d9179e) is an example of a random PR that I saw from Jan 2019, and there also the option for commit comments is not present at the end as it should have been).
![image](https://user-images.githubusercontent.com/24206326/57177025-6bcbda80-6e7d-11e9-9879-adfd4d5e3250.png)

2) I also wanted to ask regarding commenting on a specific line in the **Files changed**. Two options present : Add single line comment, and Start a review.
I suppose that when we select add single line comment, the comment is counted as a commit comment. Whereas any commit that is made after selecting the option Start a review (and before submitting the review) is counted as a review.
If this is the case, this PR can fetch the line comments (since they are counted as commit comments).